tests/v3_validation/cattlevalidationtest/core/test_hosttemplate.py

```diff
@@ -0,0 +1,550 @@
+                    client.list_hostTemplate().data))
+
+@if_machine_digocean
+def test_create_delete_do(client):
+    create_args = {
+        "name": "test-do-ht1",
+        "driver": "digitalocean",
+        "secretValues": {
+            "digitaloceanConfig": {
+                "accessToken": access_token
+            }
+        }
+    }
+    ht1 = create_ht(create_args, client)
+    cleanupht(ht1, client)
+
+@if_machine_amazon
+def test_create_delete_aws(client):
+    create_args = {
+        "name": "test-aws-ht1",
+        "driver": "amazonec2",
+        "secretValues": {
+            "amazonec2Config": {
+                "accessKey": access_key,
+                "secretKey": secret_key
+            }
+        }
+    }
+    ht1 = create_ht(create_args, client)
+    cleanupht(ht1, client)
+
+@if_machine_amazon
+@if_machine_digocean
+def test_single_provider_only():
+    create_args = [{
+        "name": random_str(),
+        "secretValues": {
+            "amazonec2Config": {
+                "accessKey": access_key,
+                "secretKey": secret_key
+            },
+            "digitaloceanConfig": {
+                "accessToken": access_token
+            }
+        }
+    }, {
+        "name": random_str(),
+        "secretValues": {
+            "digitaloceanConfig": {
+                "accessToken": access_token
+            }
+        },
+        "publicValues": {
+            "amazonec2Config": {
+                "ami": "ami-7caa341c",
+                "instanceType": "t2.micro",
+                "region": "us-west-2",
+                "rootSize": "16",
+                "securityGroup": [
+                    "rancher-machine",
+                ],
+                "sshUser": "rancher",
+                "subnetId": "subnet-e9fcc78d",
+                "volumeType": "gp2",
+                "vpcId": "vpc-08d7c46c",
+                "zone": "a"
+            }
+        }
+    }, {
+        "name": random_str(),
+        "secretValues": {
+            "digitaloceanConfig": {
+                "accessToken": access_token
+            }
+        },
+        "publicValues": {
+            "digitaloceanConfig": {
+                "image": "ubuntu-17-04-x64",
+                "region": "sfo2",
+                "size": "1gb"
+            },
+            "amazonec2Config": {
+                "ami": "ami-7caa341c",
+                "instanceType": "t2.micro",
+                "region": "us-west-2",
+                "rootSize": "16",
+                "securityGroup": [
+                    "rancher-machine",
+                ],
+                "sshUser": "rancher",
+                "subnetId": "subnet-e9fcc78d",
+                "volumeType": "gp2",
+                "vpcId": "vpc-08d7c46c",
+                "zone": "a"
+            }
+        }
+    }]
+
+    hts = []
+    for ht_args in create_args:
+        try:
+            hts.append(create_ht(ht_args, client))
+        except:
+            pass
+
+    assert [] == hts
```

Comment:
Actually now that I think about it, since this is not actually attempting to create a host in one of the cloud providers, this should be an integration test in cattle. Can you move it to here: https://github.com/rancher/cattle/blob/master/tests/integration/cattletest/core/test_host_template.py ?